### ISSUE TITLE
utxo will never get confirmed is `blockHashOpt` is not defined

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#4168]
This bug was seen on d1fc3275 using neutrino

https://github.com/bitcoin-s/bitcoin-s/blob/7a5c2971dd10f271da44a968e3089d8f9e7383b2/wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala#L120

If the `blockHashOpt` never gets defined for a `outPoint` that we have inside of `txo_spending_info`, the utxo will never get confirmed.

I'm seeing behavior in my wallet where a utxo [49388788207a1e3d9c236c5a321530d6ad4787508cb74a7ee9106967b4901b70](https://mempool.space/tx/49388788207a1e3d9c236c5a321530d6ad4787508cb74a7ee9106967b4901b70) is never getting confirmed in my wallet due to the fact my database does not have a block hash defined for the tx. This means we are somehow missing the insertion of the block_hash for a tx in my wallet

The next question is why didn't we match a compact filter for this block? This is my logs when i processed the block [0000000000000000000311c3b5633a8146f7a9798f61d1c789063d718afb772e](https://mempool.space/block/0000000000000000000311c3b5633a8146f7a9798f61d1c789063d718afb772e) which contains the tx [49388788207a1e3d9c236c5a321530d6ad4787508cb74a7ee9106967b4901b70](https://mempool.space/tx/49388788207a1e3d9c236c5a321530d6ad4787508cb74a7ee9106967b4901b70)

```
2022-03-02T13:18:52UTC INFO [DLCWallet$DLCWalletImpl] Successfully inserted UTXO 49388788207a1e3d9c236c5a321530d6ad4787508cb74a7ee9106967b4901b70:1 amt=30710250 sats into DB
2022-03-02T13:18:52UTC INFO [DLCWallet$DLCWalletImpl] Processing received utxos in tx 49388788207a1e3d9c236c5a321530d6ad4787508cb74a7ee9106967b4901b70 for 1 DLC(s)
2022-03-02T13:18:52UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 received outputs, it took=10ms
2022-03-02T13:18:52UTC INFO [DLCWallet$DLCWalletImpl] 2 txos are now confirmed!
2022-03-02T13:18:52UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 2 spent outputs, it took=5ms
2022-03-02T13:22:48UTC INFO [DataMessageHandler] Received headers message with 1 headers
2022-03-02T13:22:48UTC INFO [ChainHandler] Processed headers from height=725591 to 725592. Best hash=000000000000000000018931fe1328f524a5b7db7d06ecae9c7e860d4b710154
2022-03-02T13:22:48UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
2022-03-02T13:22:48UTC INFO [DLCWallet$DLCWalletImpl] 1 txos are now confirmed!
2022-03-02T13:22:48UTC INFO [DLCWallet$DLCWalletImpl] 3 txos are now confirmed!
2022-03-02T13:22:49UTC INFO [ChainHandler] Processed filters headers from height=725592 to 725592. Best filterheader.blockHash=000000000000000000018931fe1328f524a5b7db7d06ecae9c7e860d4b710154
2022-03-02T13:22:51UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters from startHeightOpt=Some(725591)
2022-03-02T13:22:51UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=725592, stopBlockHash=000000000000000000018931fe1328f524a5b7db7d06ecae9c7e860d4b710154)
2022-03-02T13:22:52UTC INFO [DataMessageHandler] We are synced
2022-03-02T13:22:52UTC INFO [DataMessageHandler] Processing 1 filters
2022-03-02T13:22:52UTC INFO [ChainHandler] Processed filters from height=725592 to 725592. Best filter.blockHash=000000000000000000018931fe1328f524a5b7db7d06ecae9c7e860d4b710154
2022-03-02T13:36:18UTC INFO [DataMessageHandler] Received headers message with 1 headers
2022-03-02T13:36:18UTC INFO [ChainHandler] Processed headers from height=725592 to 725593. Best hash=0000000000000000000311c3b5633a8146f7a9798f61d1c789063d718afb772e
2022-03-02T13:36:18UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
2022-03-02T13:36:18UTC INFO [DLCWallet$DLCWalletImpl] 1 txos are now confirmed!
2022-03-02T13:36:18UTC INFO [DLCWallet$DLCWalletImpl] 3 txos are now confirmed!
2022-03-02T13:36:23UTC INFO [ChainHandler] Processed filters headers from height=725593 to 725593. Best filterheader.blockHash=0000000000000000000311c3b5633a8146f7a9798f61d1c789063d718afb772e
2022-03-02T13:36:25UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters from startHeightOpt=Some(725592)
2022-03-02T13:36:25UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=725593, stopBlockHash=0000000000000000000311c3b5633a8146f7a9798f61d1c789063d718afb772e)
2022-03-02T13:36:26UTC INFO [DataMessageHandler] We are synced
2022-03-02T13:36:26UTC INFO [DataMessageHandler] Processing 1 filters
2022-03-02T13:36:26UTC INFO [ChainHandler] Processed filters from height=725593 to 725593. Best filter.blockHash=0000000000000000000311c3b5633a8146f7a9798f61d1c789063d718afb772e
2022-03-02T13:52:43UTC INFO [P2PClientActor] We've been disconnected by Peer(neutrino.suredbits.com:8333) command=PeerClosed state=Normal
//note the timestamp above, 15 minutes after processing the filter
```

<img width="1440" alt="Screen Shot 2022-03-05 at 4 21 31 PM" src="https://user-images.githubusercontent.com/3514957/156901676-ea5a1ea4-7a85-4770-aa58-27b2bffe4408.png">

**Comment by Christewart** [comment]
When rescanning the wallet, this utxo shows up as confirmed as expected. 

The question that remains is: 

Why did the utxo not get matched by `processCompactFilter()` the first time we ran this? It appears the utxo got inserted into our wallet because we receive it [via a nodecallback](https://github.com/bitcoin-s/bitcoin-s/blob/98c5d816ac9fac10da020394377d060f58110cd7/app/server/src/main/scala/org/bitcoins/server/util/CallbackUtil.scala#L24). It somehow got updated to `PendingConfirmationsReceived` which i don't understand. It should be `BroadcastReceived`. 

<img width="1440" alt="Screen Shot 2022-03-06 at 12 19 54 PM" src="https://user-images.githubusercontent.com/3514957/156936515-37676551-eb02-432c-b468-635e880b86d0.png">

**Comment by Christewart** [comment]
When rescanning the wallet, this utxo shows up as confirmed as expected. 

The question that remains is: 

Why did the utxo not get matched by `processCompactFilter()` the first time we ran this? It appears the utxo got inserted into our wallet because we receive it [via a nodecallback](https://github.com/bitcoin-s/bitcoin-s/blob/98c5d816ac9fac10da020394377d060f58110cd7/app/server/src/main/scala/org/bitcoins/server/util/CallbackUtil.scala#L24). It somehow got updated to `PendingConfirmationsReceived` which i don't understand. It should be `BroadcastReceived`. 

<img width="1440" alt="Screen Shot 2022-03-06 at 12 19 54 PM" src="https://user-images.githubusercontent.com/3514957/156936515-37676551-eb02-432c-b468-635e880b86d0.png">
